modules/xfem/include/dirackernels/XFEMInterfaceKernel.h

```diff
@@ -0,0 +1,46 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// Moose Includes
+#include "DiracKernel.h"
+#include "GeometricCutUserObject.h"
+
+class XFEMInterfaceKernel : public DiracKernel
```

Comment:
In MOOSE, `interfaceKernel` is used to couple different physics on the interface, so it needs element and its neighbor information. That is more like the 'XFEMConstraint`. Here, the new derived class is mainly for boundary conditions on either side of the interface, so I suggest we use a different name here to avoid any confusion. What do you think?

Reply:
Right. I renamed it to `XFEMIntegratedBC`.